glib-macros/src/closure.rs

```diff
@@ -222,6 +224,39 @@ impl ToTokens for Closure {
             .iter()
             .map(|c| c.outer_after_tokens(&crate_ident, &closure_ident));
 
+        let closure = &self.closure;
+
+        let arg_and_return_types = {
+            let type_parameter_idents = self
+                .args
+                .iter()
+                .enumerate()
+                .map(|(index, _arg)| Ident::new(&format!("T{}", index), Span::call_site()))
+                .collect::<Vec<_>>();
+
+            quote! {
+                let #get_types_ident: Box<dyn Fn(Option<&_>) -> _ + Send + Sync> = Box::new(|f| {
```

Comment:
This is all a bit convoluted but I couldn't find another way. The `Box` is needed so that we can move this very closure into our other closure below in a meaningful way. But I think it should be possible without a `Box` somehow.

Reply:
Should probably also move this block into the other one at the bottom. No advantage of having it separate.

Reply:
Got rid of the box and simplified this part of the code a bit.